<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>BarcodeObj object — BarcodeObj-class • CellBarcode</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="BarcodeObj object — BarcodeObj-class"><meta property="og:description" content="A S4 object holds the barcode data and samples' metadata. A set of operations
can be applied to the BarcodeObj object for quality control and selecting
barcodes/samples subset."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellBarcode</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Barcode_in_10X_scRNASeq.html">Get barcode from 10X Genomics scRNASeq data</a>
    </li>
    <li>
      <a href="../articles/UMI_VDJ_Barcode.html">Analyzing Cellular DNA Barcode with CellBarcode</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>BarcodeObj object</h1>
    
    <div class="hidden name"><code>BarcodeObj.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A S4 object holds the barcode data and samples' metadata. A set of operations
can be applied to the BarcodeObj object for quality control and selecting
barcodes/samples subset.</p>
    </div>


    <div id="details">
    <h2>Details</h2>
    <p>The BarcodeObj object is a S4 object, it has three slots, 
which can be access by "@" operator, they are <code>messyBc</code>, <code>cleanBc</code> and
<code>metadata</code>. A <code>BarcodeObj</code> object can be generated by <code>bc_extract</code>
function. The <code>bc_extract</code> function can use various data types as input,
such as data.frame, fastq files or ShortReadQ.</p>
<p>Slot <code>messyBc</code> is a list holds the raw barcodes sequence without filtering,
where each element is a <code>data.table</code> corresponding to the successive samples.
Each table has 3 columns: 1. <code>umi_seq</code> (optional): UMI sequence. 2.
<code>barcode_seq</code>: barcode sequence. 3. <code>count</code>: how many reads a full sequence
has. In this table, <code>barcode_seq</code> value can be duplicated, as two different
full read sequences can have the same barcode sequence, due to the
diversity of the UMI or mutations in the constant region.</p>
<p>Slot <code>cleanBc</code> is a <code>list</code> holds the barcodes sequence after filtering,
where each element is a <code>data.table</code> corresponding to the successive samples.
The "cleanBc" slot contains 2 columns 1. <code>barcode_seq</code>: barcode sequence
2. <code>counts</code>: reads count, or UMI count if the <code>cleanBc</code> was created by
<code>bc_cure_umi</code>.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#######</span></span></span>
<span class="r-in"><span><span class="co"># Create BarcodeObj with fastq file</span></span></span>
<span class="r-in"><span><span class="va">fq_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"simple.fq"</span>, package<span class="op">=</span><span class="st">"CellBarcode"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/ShortRead" class="external-link">ShortRead</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocParallel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biostrings</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:plyr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rename</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:data.table’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     first, second</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘IRanges’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:plyr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     desc</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:data.table’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     shift</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: XVector</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘XVector’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:plyr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     compact</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Rsamtools</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘GenomicRanges’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:magrittr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     subtract</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicAlignments</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘matrixStats’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:plyr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     count</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biobase’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:MatrixGenerics’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘GenomicAlignments’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:data.table’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     last</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘ShortRead’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:plyr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     id</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:magrittr’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     functions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:data.table’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tables</span>
<span class="r-in"><span><span class="fu"><a href="bc_extract.html">bc_extract</a></span><span class="op">(</span><span class="va">fq_file</span>, pattern <span class="op">=</span> <span class="st">"AAAAA(.*)CCCCC"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    barcode_seq count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;char&gt; &lt;int&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1:          GG     1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#######</span></span></span>
<span class="r-in"><span><span class="co"># data manipulation on BarcodeObj object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bc_obj</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 3 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 9 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 5 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select barcodes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="bc_subset.html">bc_subset</a></span><span class="op">(</span><span class="va">bc_obj</span>, barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AACCTT"</span>, <span class="st">"AACCTT"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 3 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 0 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 0 barcodes </span>
<span class="r-in"><span><span class="va">bc_obj</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AGAG"</span>, <span class="st">"AAAG"</span><span class="op">)</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 3 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 6 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 5 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 2 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 2 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select samples by meta data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="bc_meta.html">bc_meta</a></span><span class="op">(</span><span class="va">bc_obj</span><span class="op">)</span><span class="op">$</span><span class="va">phenotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"l"</span>, <span class="st">"b"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="bc_meta.html">bc_meta</a></span><span class="op">(</span><span class="va">bc_obj</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       raw_read_count barcode_read_count depth_cutoff phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> test1            184                178            5         l</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> test2            124                118            5         b</span>
<span class="r-in"><span><span class="fu"><a href="bc_subset.html">bc_subset</a></span><span class="op">(</span><span class="va">bc_obj</span>, sample <span class="op">=</span> <span class="va">phenotype</span> <span class="op">==</span> <span class="st">"l"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 1 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 1 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select samples by sample name</span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span><span class="op">[</span>, <span class="st">"test1"</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 1 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 1 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes </span>
<span class="r-in"><span><span class="va">bc_obj</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"test1"</span>, <span class="st">"test2"</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 9 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 5 barcodes </span>
<span class="r-in"><span><span class="fu"><a href="bc_subset.html">bc_subset</a></span><span class="op">(</span><span class="va">bc_obj</span>, sample <span class="op">=</span> <span class="st">"test1"</span>, barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AACCTT"</span>, <span class="st">"AACCTT"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 1 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 1 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Apply barcodes black list</span></span></span>
<span class="r-in"><span><span class="fu"><a href="bc_subset.html">bc_subset</a></span><span class="op">(</span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span>,</span></span>
<span class="r-in"><span>    sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"test1"</span>, <span class="st">"test2"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    barcode <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AACCTT"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 0 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 0 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 0 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Join two samples with no barcodes overlap</span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test1"</span><span class="op">]</span> <span class="op">+</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AAAG"</span>, <span class="st">"test2"</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 8 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  phenotype.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AAAG", "test2"]  barcode_read_count.bc_obj["AAAG", "test2"]  depth_cutoff.bc_obj["AAAG", "test2"]  phenotype.bc_obj["AAAG", "test2"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 1 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 1 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Join two samples with barcodes overlap</span></span></span>
<span class="r-in"><span><span class="va">bc_obj_join</span> <span class="op">&lt;-</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test1"</span><span class="op">]</span> <span class="op">+</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test2"</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">bc_obj_join</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 8 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  phenotype.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AGAG", "test2"]  barcode_read_count.bc_obj["AGAG", "test2"]  depth_cutoff.bc_obj["AGAG", "test2"]  phenotype.bc_obj["AGAG", "test2"]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 1 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 1 barcodes </span>
<span class="r-in"><span><span class="co"># The same barcode will be merged after applying bc_cure_depth()</span></span></span>
<span class="r-in"><span><span class="fu"><a href="bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_join</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ------------</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> bc_cure_depth: isUpdate is TRUE, update the cleanBc.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 9 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  phenotype.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AGAG", "test2"]  barcode_read_count.bc_obj["AGAG", "test2"]  depth_cutoff.bc_obj["AGAG", "test2"]  phenotype.bc_obj["AGAG", "test2"]  depth_cutoff</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 1 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 1 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Remove barcodes</span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 9 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 5 barcodes </span>
<span class="r-in"><span><span class="va">bc_obj</span> <span class="op">-</span> <span class="st">"AAAG"</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 7 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 7 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 3 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 4 barcodes </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Select barcodes in a white list</span></span></span>
<span class="r-in"><span><span class="va">bc_obj</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 10 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 9 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 4 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 5 barcodes </span>
<span class="r-in"><span><span class="va">bc_obj</span> <span class="op">*</span> <span class="st">"AAAG"</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 4 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count  depth_cutoff  phenotype</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 2 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 3 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @cleanBc: 2 samples for cleaned barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test1 there are: 1 barcodes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $test2 there are: 1 barcodes </span>
<span class="r-in"><span><span class="co">###</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Wenjie Sun, Anne-Marie Lyne, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

