<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Parse 10X Genomic scRNASeq sam file (Experimental) — bc_extract_10XscSeq • CellBarcode</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Parse 10X Genomic scRNASeq sam file (Experimental) — bc_extract_10XscSeq"><meta property="og:description" content="bc_extract_10XscSeq can extract cellular barcode, UMI and lineage barcode
sequences from 10X Genomics scRNASeq bam file. This function can not process
bam file directly, user needs to uncompress the bam file to get sam file in
order to run this function See example."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-NY9M0DHR4F"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-NY9M0DHR4F');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellBarcode</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Barcode_in_10X_scRNASeq.html">Get barcode from 10X Genomics scRNASeq data</a>
    </li>
    <li>
      <a href="../articles/UMI_VDJ_Barcode.html">Analyzing Cellular DNA Barcode with CellBarcode</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Parse 10X Genomic scRNASeq sam file (Experimental)</h1>
    
    <div class="hidden name"><code>bc_extract_10X_scSeq.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bc_extract_10XscSeq</code> can extract cellular barcode, UMI and lineage barcode
sequences from 10X Genomics scRNASeq bam file. This function can not process
bam file directly, user needs to uncompress the bam file to get sam file in
order to run this function See example.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bc_extract_10XscSeq</span><span class="op">(</span><span class="va">sam</span>, <span class="va">pattern</span>, cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>, umi_tag <span class="op">=</span> <span class="st">"UR"</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>sam</dt>
<dd><p>A string, define the un-mapped sequences</p></dd>


<dt>pattern</dt>
<dd><p>A string, define the regular expression to match the barcode
sequence. The barcode sequence should be in the first catch. Please see the
documents of <code><a href="bc_extract.html">bc_extract</a></code> and example for more information.</p></dd>


<dt>cell_barcode_tag</dt>
<dd><p>A string, define the tag of 10X cell barcode field in sam
file. The default is "CR".</p></dd>


<dt>umi_tag</dt>
<dd><p>A string, define the tag of UMI field in the sam file.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data.frame with 4 columns:</p><ol><li><p><code>cell_barcode</code>: 10X cellular barcode.</p></li>
<li><p><code>umi</code>: UMI sequence.</p></li>
<li><p><code>barcode_seq</code>: lineage barcode.</p></li>
<li><p><code>count</code>: reads count.</p></li>
</ol></div>
    <div id="details">
    <h2>Details</h2>
    <p>If the barcode sequence does not map to the reference genome. The user should
use samtools to get the un-mapped reads and save it as sam format for using
as the input. It can save a lot of time. The way to get the un-mapped reads:</p><div class="sourceCode"><pre><code>
samtools view -f 4 input.bam &gt; output.sam 
</code></pre></div>

    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## NOT run</span></span></span>
<span class="r-in"><span><span class="co"># In the case that when the barcode sequence is not mapped to </span></span></span>
<span class="r-in"><span><span class="co"># reference genome, it will be much more efficiency to get </span></span></span>
<span class="r-in"><span><span class="co"># the un-mapped sequences as the input.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get un-mapped reads</span></span></span>
<span class="r-in"><span><span class="co"># samtools view -f 4 input.bam &gt; scRNASeq_10X.sam </span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNASeq_10X.sam"</span>, package <span class="op">=</span> <span class="st">"CellBarcode"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">bc_extract_10XscSeq</span><span class="op">(</span></span></span>
<span class="r-in"><span>  sam <span class="op">=</span> <span class="va">sam_file</span>,</span></span>
<span class="r-in"><span>  pattern <span class="op">=</span> <span class="st">"AGATCAG(.*)TGTGGTA"</span>,</span></span>
<span class="r-in"><span>  cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>,</span></span>
<span class="r-in"><span>  umi_tag <span class="op">=</span> <span class="st">"UR"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        cell_barcode          umi           barcode_seq count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  ACAACCATCACGGAGA GCTGGGATCCGG GGGAGTGGTGGAAGGTTAGGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  AATCGACTCATCGCTC CTTTTGTATGGA CCTCGGCAGAGTACTTTTTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  AAGTTCGAGAGGTATT CCACGGGCACCG CGTTAATCGGTCTAGTGATTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  AAGTCGTAGCACGGAT AAAGCCTGCAAG CTATATATTCGTTGTATTATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  AAGACAAAGGCGCTTC CGTCTTGGTAGC TCATTTGATACTAAGAATGAA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  AACTTCTGTGTTCCTC GACAACCTTGGT GGGTGGGATTCAGTGATCATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  AACCTTTGTCATTCCC TAATAATTGCGT CTTAACATATGTAATATCTCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  AACGAAATCTGGGTCG CCAGAAACGTAC GCGGCTTCATTCTAGTGCCAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  AAAGGTAAGGCCCACT AGAATTTTGTCA         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 AAGCAGTGGTATCAAC GCAGAGTACATG CTATATATTCGTTGTATTATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 11 AAAGGATAGTGTGTTC ATCGTTCAAGTA CGTTAATCGGTCTAGTGATTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12 AAAGGGCAGCCTGAGA GAAAGTTGAGTA CGTCGATTGTGCAAACGTAGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 13 TTCACCGATTTTGTAA TCCAGAGGTTGA TTTATCGCTTGATTTTGTATA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 14 AGGCCTCACATTCTTC TCTACCATACAT AGTGCTTTTGCTGGGTGGTGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 15 CACAAATTTTGTAATC CAGAGGTTGATT GCGGGATCACTCGCAGGGTCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 16 TTCGCTGATTTTGTAA TCCAGAGGTTGA AGACAACTTGGATTTTGTGTA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17 AAACGCTAGCTGACCC CACGCTTATGTT TACCTAAGGTCGAATTTGCTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 18 TAAGCCAAAAGAACAA AGCTGGAGGCAT         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 19 AGCAGTGGTATCAACG CAGAGTACATGG ATTGGACCATGTAATGAGTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 20 AAACGAAAGTTCATGC TGGTTCTGTTTC ACACATTGTTATACGTAATAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 21 AAGGAGCTGACAGGTG GTGGCAATGCCC           TACTAGCGTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 22 TGAAAGTGACAACTGA GTACAGACGATG ATAGGGCATCAGAATAGTTTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 23 AGTCTTTCGTCAAACA CGCACGACAATT         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 24 TTCACCGATTTTGTAA TCCAGAGGTTGA TCATGTTTCTGCTTCATTCCA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 25 AACACACCAATCCAGT CGGAGAGATTAC CCTACTTTAATTGAGTCAATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 26 GGAAGCAATAGCATGA TACAAAGGCATT TAACAAACTTGACGGCCTTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 27 ACAAAGAGTAGGTCAG ATCTTCAAATAT                    CC     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 28 AAGTGAAAGTCCCGGT AGCCATTGTCCT TTTAATAGTTTCTATTTTGGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 29 GCAGTGGTATCAACGC AGAGTACATGGG TTTATCGCTTGATTTTGTATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 30 AAGCAGTGGTATCAAC GCAGAGTACATG ATTGGACCATGTAATGAGTAA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31 TAAGCCATAAACATAT TTAATAATATAT       CATGGCAAAACCAGC     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 32 AAGCAGTGGTATCAAC GCAGAGTACATG TGTTCACTAGTCATAGCGACA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 33 CAAATTTTGTAATCCA GAGGTTGATTGT CTAAAGATAGTATGACTAGTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 34 AAGCAGTGGTATCAAC GCAGAGTACATG           TAGGGTTTCAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 35 AAGCAGTGGTATCAAC GCAGAGTACATG ATTGGTTAGAGTTCTTGATCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 36 AGGCATTAAAGCAGCG TATCCACATAGC  GCGTTGTTGAAATTACGTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 37 AATAGAGGTTCTAACG AAGACAGGCTTA AGCCATTATGTCGTGTCATGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 38 GTGGTATCAACGCAGA GTACATGGGAGC TGCTTTATGTATGTAGTGGAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 39 TTCTTCACAGAATTGG AAAAAACTACTT         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 40 AAGACAAAGCTAAATG GTATTCTCTGAG         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 41 AAGCAGTGGTATCAAC GCAGAGTACATG AGCTTTATCCTAGTTGTCGCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 42 AACACACTCACCGACG CTGCTCTAGGAG CGCGATCGTAGGTCCGAGTCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 43 AAGCAGTGGTATCAAC GCAGAGTACATG GGGAGTGGTGGAAGGTTAGGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 44 ACATGGGGTAAAAGGA GCAACATAGTTA CTGGTTTCTCGTTCCAATTAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45 GCAGTGGTATCAACGC AGAGTACATGGG TACCAGAGCACCTTATTTGTA     2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 46 CAGTGGTATCAACGCA GAGTACATGGGG TTTTGGATATTGTCCCCTAAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47 AAGCAGTGGTATCAAC GCAGAGTACATG  TGTGCTTAGCAGAGGGATGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 48 GTTAAGAATACCAGTC AATCTTTCACAA TGCGTTTAAAGGCGGCTGGGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 49 CACAACTCCTCATAAA GAGACAGCAACC CCTTTTTATTAGATCACCACA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 50 AACCCAAGGTAATTTA TAGATTCAATGC         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 51 GTGGTATCAACGCAGA GTACATGGGGCA GACAAGGGTTGGGGGACATCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 52 AAAGTGATCTGACGCG CATCAATGACTG TATCGTTCGTTTATGTACCCA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 53 TTGGGCGTCTTTGGGC GGCCGCGTGATC AGCCATTATGTCGTGTCATGA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 54 CTACGGGAAGCAATAG CATGATACAAAG TACCTAAGGTCGAATTTGCTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 55 TTCCTCTCAGAATTGG AAAAAATGACTT         ATGGTTGTAGATG     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 56 TTCCAAATTTTGTAAT CCAGAGGTTGAT CATGCATCCAGAATTGATATA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 57 CCTTGTGAGTGTTACC ATCTTGACCCGT CATATCGTACGGTGTTAGTTA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 58 GATACAAAGGCATTAA AGCACCGTATCC TCCTATTCTGTTGGAGTACAA     1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 59 AAAGTCCCACTGCGAC TCGGAACCGCGT AGACAACTTGGATTTTGTGTA     1</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Wenjie Sun, Anne-Marie Lyne, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

