<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Extract barcode from single-cell sequencing sam file — bc_extract_sc_sam • CellBarcode</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Extract barcode from single-cell sequencing sam file — bc_extract_sc_sam"><meta property="og:description" content="bc_extract_sc_sam can extract cellular barcode, UMI, and lineage
barcode sequences from 10X Genomics scRNASeq sam file (or bam file have
similar data structure). This function can not process bam file directly,
users need to uncompress the bam file to get a sam file to run this
function See example."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-NY9M0DHR4F"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-NY9M0DHR4F');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellBarcode</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Barcode_in_10X_scRNASeq.html">Get barcode from 10X Genomics scRNASeq data</a>
    </li>
    <li>
      <a href="../articles/UMI_VDJ_Barcode.html">Analyzing Cellular DNA Barcode with CellBarcode</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Extract barcode from single-cell sequencing sam file</h1>
    
    <div class="hidden name"><code>bc_extract_sc_sam.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bc_extract_sc_sam</code> can extract cellular barcode, UMI, and lineage
barcode sequences from 10X Genomics scRNASeq sam file (or bam file have
similar data structure). This function can not process bam file directly,
users need to uncompress the bam file to get a sam file to run this
function See example.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">bc_extract_sc_sam</span><span class="op">(</span><span class="va">sam</span>, <span class="va">pattern</span>, cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>, umi_tag <span class="op">=</span> <span class="st">"UR"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">bc_extract_sc_bam</span><span class="op">(</span><span class="va">bam</span>, <span class="va">pattern</span>, cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>, umi_tag <span class="op">=</span> <span class="st">"UR"</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>sam</dt>
<dd><p>A string, define the un-mapped sequences</p></dd>


<dt>pattern</dt>
<dd><p>A string, define the regular expression to match the barcode
sequence. The barcode sequence should be in the first catch. Please see the
documents of <code><a href="bc_extract.html">bc_extract</a></code> and example for more information.</p></dd>


<dt>cell_barcode_tag</dt>
<dd><p>A string, define the tag of cellular barcode field in sam
file. The default is "CR".</p></dd>


<dt>umi_tag</dt>
<dd><p>A string, define the tag of a UMI field in the sam file.</p></dd>


<dt>bam</dt>
<dd><p>A string, define the bam file, it will be converted to sam file</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A BarcodeObj object with each cell as a sample.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Although the function `bc_extract_sc_bam` can process bam file directly,
some optimization is still working on, it will be much more efficient to use
`samtools` to get the sam file.</p>
<p>What's more, if the barcode sequence does not map to the reference genome. The user should
use the samtools to get the un-mapped reads and save it as sam format for using
as the input. It can save a lot of time. The way to get the un-mapped reads:</p><div class="sourceCode"><pre><code>
samtools view -f 4 input.bam &gt; output.sam 
</code></pre></div>

    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="bc_extract.html">bc_extract</a></code>,
<code><a href="bc_extract_sc_fastq.html">bc_extract_sc_fastq</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## NOT run</span></span></span>
<span class="r-in"><span><span class="co"># In the case that when the barcode sequence is not mapped to </span></span></span>
<span class="r-in"><span><span class="co"># reference genome, it will be much more efficient to get </span></span></span>
<span class="r-in"><span><span class="co"># the un-mapped sequences as the input.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get un-mapped reads</span></span></span>
<span class="r-in"><span><span class="co"># samtools view -f 4 input.bam &gt; scRNASeq_10X.sam </span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">sam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNASeq_10X.sam"</span>, package <span class="op">=</span> <span class="st">"CellBarcode"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">bc_extract_sc_sam</span><span class="op">(</span></span></span>
<span class="r-in"><span>  sam <span class="op">=</span> <span class="va">sam_file</span>,</span></span>
<span class="r-in"><span>  pattern <span class="op">=</span> <span class="st">"AGATCAG(.*)TGTGGTA"</span>,</span></span>
<span class="r-in"><span>  cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>,</span></span>
<span class="r-in"><span>  umi_tag <span class="op">=</span> <span class="st">"UR"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 2 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 49 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAACGAAAGTTCATGC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAACGCTAGCTGACCC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGATAGTGTGTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGGCAGCCTGAGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGTAAGGCCCACT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGTCCCACTGCGAC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGTGATCTGACGCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACACACCAATCCAGT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACACACTCACCGACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACCCAAGGTAATTTA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACCTTTGTCATTCCC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACGAAATCTGGGTCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACTTCTGTGTTCCTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGACAAAGCTAAATG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGACAAAGGCGCTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGCAGTGGTATCAAC there are: 8 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGGAGCTGACAGGTG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTCGTAGCACGGAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTGAAAGTCCCGGT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTTCGAGAGGTATT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AATAGAGGTTCTAACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AATCGACTCATCGCTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACAAAGAGTAGGTCAG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACAACCATCACGGAGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACATGGGGTAAAAGGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGCAGTGGTATCAACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGGCATTAAAGCAGCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGGCCTCACATTCTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGTCTTTCGTCAAACA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CAAATTTTGTAATCCA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CACAAATTTTGTAATC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CACAACTCCTCATAAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CAGTGGTATCAACGCA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CCTTGTGAGTGTTACC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CTACGGGAAGCAATAG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GATACAAAGGCATTAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GCAGTGGTATCAACGC there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GGAAGCAATAGCATGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GTGGTATCAACGCAGA there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GTTAAGAATACCAGTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TAAGCCAAAAGAACAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TAAGCCATAAACATAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TGAAAGTGACAACTGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCACCGATTTTGTAA there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCCAAATTTTGTAAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCCTCTCAGAATTGG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCGCTGATTTTGTAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCTTCACAGAATTGG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTGGGCGTCTTTGGGC there are: 1 Tags </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Read bam file directly</span></span></span>
<span class="r-in"><span><span class="va">bam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNASeq_10X.bam"</span>, package <span class="op">=</span> <span class="st">"CellBarcode"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">bc_extract_sc_bam</span><span class="op">(</span></span></span>
<span class="r-in"><span>   bam <span class="op">=</span> <span class="va">bam_file</span>,</span></span>
<span class="r-in"><span>   pattern <span class="op">=</span> <span class="st">"AGATCAG(.*)TGTGGTA"</span>,</span></span>
<span class="r-in"><span>   cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>,</span></span>
<span class="r-in"><span>   umi_tag <span class="op">=</span> <span class="st">"UR"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Start to convert bam file to sam file.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> sam file path:  /var/folders/46/vh_b6qzs5kzgdhvx1kcn7hyh0000gn/T//RtmpdbkBWL/output.sam </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Bonjour le monde, This is a BarcodeObj.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> It contains: </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @metadata: 2 field(s) available:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> raw_read_count  barcode_read_count</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ----------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> @messyBc: 49 sample(s) for raw barcodes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAACGAAAGTTCATGC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAACGCTAGCTGACCC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGATAGTGTGTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGGCAGCCTGAGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGGTAAGGCCCACT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGTCCCACTGCGAC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAAGTGATCTGACGCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACACACCAATCCAGT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACACACTCACCGACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACCCAAGGTAATTTA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACCTTTGTCATTCCC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACGAAATCTGGGTCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AACTTCTGTGTTCCTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGACAAAGCTAAATG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGACAAAGGCGCTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGCAGTGGTATCAAC there are: 8 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGGAGCTGACAGGTG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTCGTAGCACGGAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTGAAAGTCCCGGT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AAGTTCGAGAGGTATT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AATAGAGGTTCTAACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AATCGACTCATCGCTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACAAAGAGTAGGTCAG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACAACCATCACGGAGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $ACATGGGGTAAAAGGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGCAGTGGTATCAACG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGGCATTAAAGCAGCG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGGCCTCACATTCTTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $AGTCTTTCGTCAAACA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CAAATTTTGTAATCCA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CACAAATTTTGTAATC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CACAACTCCTCATAAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CAGTGGTATCAACGCA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CCTTGTGAGTGTTACC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $CTACGGGAAGCAATAG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GATACAAAGGCATTAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GCAGTGGTATCAACGC there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GGAAGCAATAGCATGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GTGGTATCAACGCAGA there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $GTTAAGAATACCAGTC there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TAAGCCAAAAGAACAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TAAGCCATAAACATAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TGAAAGTGACAACTGA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCACCGATTTTGTAA there are: 2 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCCAAATTTTGTAAT there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCCTCTCAGAATTGG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCGCTGATTTTGTAA there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTCTTCACAGAATTGG there are: 1 Tags</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     In sample $TTGGGCGTCTTTGGGC there are: 1 Tags </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Wenjie Sun, Anne-Marie Lyne, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

