<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Get barcode from 10X Genomics scRNASeq data • CellBarcode</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Get barcode from 10X Genomics scRNASeq data">
<meta property="og:description" content="CellBarcode">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-NY9M0DHR4F"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-NY9M0DHR4F');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellBarcode</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Barcode_in_10X_scRNASeq.html">Get barcode from 10X Genomics scRNASeq data</a>
    </li>
    <li>
      <a href="../articles/UMI_VDJ_Barcode.html">Analyzing Cellular DNA Barcode with CellBarcode</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Get barcode from 10X Genomics scRNASeq data</h1>
                        <h4 data-toc-skip class="author">Wenjie Sun</h4>
                        
      
      
      <div class="hidden name"><code>Barcode_in_10X_scRNASeq.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CellBarcode</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="preprocess-of-cellranger-output-bam-file">Preprocess of CellRanger output bam file<a class="anchor" aria-label="anchor" href="#preprocess-of-cellranger-output-bam-file"></a>
</h2>
<p><strong>The the information in the bam file</strong>. There are RNA
sequence, Cell barcode and UMI in the bam file. We need to get the
barcode in the RNA sequence together with the Cell barcode and UMI.</p>
<p><strong>Where to find the bam file</strong>. Usually, the bam file in
in following location of the CellRanger output.</p>
<p><code>CellRanger Output fold/outs/possorted_genome_bam.bam</code></p>
<p><strong>Why preprocess</strong>. We need get the sam file as input.
And in some cases we can do some filtering to make the input file
smaller, for reducing the running time.</p>
<p><strong>Example 1 get the sam file</strong></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view possorted_genome_bam.bam <span class="op">&gt;</span> scRNASeq_10X.sam</span></code></pre></div>
<p><strong>Example 2 get the sam file only contain un mapped
reads</strong></p>
<p>In most of the time, the barcodes are designed not overlap with the
genome sequence, so those barcodes sequences are not mapped to the
reference genome. Add a simple parameter, we can get the un-mapped reads
to significantly reduce the running time of the barcode extraction
procedure. In the following example, the <code>scRNASeq_10X.sam</code>
file only contains the un-mapped reads.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> view <span class="at">-f</span> 4 possorted_genome_bam.bam <span class="op">&gt;</span> scRNASeq_10X.sam</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="extract-lineage-barcode">Extract lineage barcode<a class="anchor" aria-label="anchor" href="#extract-lineage-barcode"></a>
</h2>
<p>Extract lineage barcode with cell-barcode, UMI information.</p>
<p>The parameters:</p>
<ul>
<li><p>sam: The location of sam format file derived from CellRanger
output bam file.</p></li>
<li><p>pattern: A regular expression, that matchs the lineage barcode.
The first capture will be extract as lineage barcode. Please check next
section to know more about how to define a pattern.</p></li>
<li><p>cell_barcode_tag: The cell barcode field tag in the sam file, the
default is “CR” in the Cell Ranger output. You do not need to modify it,
unless you know what you are doing.</p></li>
<li><p>umi_tag: The UMI field tag in the sam file, the default is “UR”
in the Cell Ranger output. You do not need to modify it, unless you know
what you are doing.</p></li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sam_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"scRNASeq_10X.sam"</span>, package <span class="op">=</span> <span class="st">"CellBarcode"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">d</span> <span class="op">=</span> <span class="fu"><a href="../reference/bc_extract_sc_sam.html">bc_extract_sc_sam</a></span><span class="op">(</span></span>
<span>   sam <span class="op">=</span> <span class="va">sam_file</span>,</span>
<span>   pattern <span class="op">=</span> <span class="st">"AGATCAG(.*)TGTGGTA"</span>,</span>
<span>   cell_barcode_tag <span class="op">=</span> <span class="st">"CR"</span>,</span>
<span>   umi_tag <span class="op">=</span> <span class="st">"UR"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">d</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 2 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 49 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $AAACGAAAGTTCATGC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAACGCTAGCTGACCC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAAGGATAGTGTGTTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAAGGGCAGCCTGAGA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAAGGTAAGGCCCACT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAAGTCCCACTGCGAC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAAGTGATCTGACGCG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACACACCAATCCAGT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACACACTCACCGACG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACCCAAGGTAATTTA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACCTTTGTCATTCCC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACGAAATCTGGGTCG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AACTTCTGTGTTCCTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGACAAAGCTAAATG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGACAAAGGCGCTTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGCAGTGGTATCAAC there are: 8 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGGAGCTGACAGGTG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGTCGTAGCACGGAT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGTGAAAGTCCCGGT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AAGTTCGAGAGGTATT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AATAGAGGTTCTAACG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AATCGACTCATCGCTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $ACAAAGAGTAGGTCAG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $ACAACCATCACGGAGA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $ACATGGGGTAAAAGGA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AGCAGTGGTATCAACG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AGGCATTAAAGCAGCG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AGGCCTCACATTCTTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $AGTCTTTCGTCAAACA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CAAATTTTGTAATCCA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CACAAATTTTGTAATC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CACAACTCCTCATAAA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CAGTGGTATCAACGCA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CCTTGTGAGTGTTACC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $CTACGGGAAGCAATAG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $GATACAAAGGCATTAA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $GCAGTGGTATCAACGC there are: 2 Tags</span></span>
<span><span class="co">#&gt;     In sample $GGAAGCAATAGCATGA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $GTGGTATCAACGCAGA there are: 2 Tags</span></span>
<span><span class="co">#&gt;     In sample $GTTAAGAATACCAGTC there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TAAGCCAAAAGAACAA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TAAGCCATAAACATAT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TGAAAGTGACAACTGA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTCACCGATTTTGTAA there are: 2 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTCCAAATTTTGTAAT there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTCCTCTCAGAATTGG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTCGCTGATTTTGTAA there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTCTTCACAGAATTGG there are: 1 Tags</span></span>
<span><span class="co">#&gt;     In sample $TTGGGCGTCTTTGGGC there are: 1 Tags</span></span></code></pre></div>
<p>The output is a <code>data.frame</code>. It contains 3 columns:</p>
<ul>
<li><p>cell_barcode: It is unique identificaiton for each cell. One
<code>cell_barcode</code> corresponding to a cell.</p></li>
<li><p>umi: It is used to label the molecular before PCR, a unique UMI
means one molecular.</p></li>
<li><p>barcode_seq: Depends on the experiment design, it usually labels
the lineage.</p></li>
<li><p>count: The reads number of a specifc cell-barcode, UMI, lineage
barcode combination.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="more-about-the-pattern">More about the <code>pattern</code><a class="anchor" aria-label="anchor" href="#more-about-the-pattern"></a>
</h2>
<p>The <code>pattern</code> is a regular expression, it tells the
function where to find the barcode. In the pattern, we define the
barcode backbone, and label the barcode sequence by bracket
<code>()</code>.</p>
<p>For example, the pattern <code>ATCG(.{21})TCGG</code> tells the
barcode is surrounded by constant sequence of <code>ATCG</code>, and
<code>TCGG</code>. Following are some examples to define the constant
region and barcode sequence.</p>
<p><strong>Example 1</strong></p>
<p><code>ATCG(.{21})</code></p>
<p>21 bases barcode after a constant sequence of “ATCG”.</p>
<p><strong>Example 2</strong></p>
<p><code>(.{15})TCGA</code></p>
<p>15 bases barcode before a constant sequence of “TCGA”.</p>
<p><strong>Example 3</strong></p>
<p><code>ATCG(.*)TCGA</code></p>
<p>A flexible length barcode between constant regions of “ATCG” and
“TCGA”.</p>
<p><strong>Need more helps</strong>: About more complex barcode pattern,
please ask the package author, then the exmaple will be apear here.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS 14.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Paris</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] CellBarcode_1.7.0 ggplot2_3.4.2     data.table_1.14.8 BiocStyle_2.28.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0            dplyr_1.1.2                </span></span>
<span><span class="co">#&gt;  [3] Biostrings_2.68.1           bitops_1.0-7               </span></span>
<span><span class="co">#&gt;  [5] fastmap_1.1.1               RCurl_1.98-1.12            </span></span>
<span><span class="co">#&gt;  [7] GenomicAlignments_1.36.0    egg_0.4.5                  </span></span>
<span><span class="co">#&gt;  [9] digest_0.6.31               lifecycle_1.0.3            </span></span>
<span><span class="co">#&gt; [11] magrittr_2.0.3              compiler_4.3.0             </span></span>
<span><span class="co">#&gt; [13] rlang_1.1.1                 sass_0.4.6                 </span></span>
<span><span class="co">#&gt; [15] tools_4.3.0                 utf8_1.2.3                 </span></span>
<span><span class="co">#&gt; [17] yaml_2.3.7                  knitr_1.43                 </span></span>
<span><span class="co">#&gt; [19] S4Arrays_1.0.4              interp_1.1-4               </span></span>
<span><span class="co">#&gt; [21] DelayedArray_0.26.3         plyr_1.8.8                 </span></span>
<span><span class="co">#&gt; [23] RColorBrewer_1.1-3          ShortRead_1.58.0           </span></span>
<span><span class="co">#&gt; [25] BiocParallel_1.34.2         withr_2.5.0                </span></span>
<span><span class="co">#&gt; [27] purrr_1.0.1                 hwriter_1.3.2.1            </span></span>
<span><span class="co">#&gt; [29] BiocGenerics_0.46.0         desc_1.4.2                 </span></span>
<span><span class="co">#&gt; [31] grid_4.3.0                  stats4_4.3.0               </span></span>
<span><span class="co">#&gt; [33] fansi_1.0.4                 latticeExtra_0.6-30        </span></span>
<span><span class="co">#&gt; [35] Ckmeans.1d.dp_4.3.4         colorspace_2.1-0           </span></span>
<span><span class="co">#&gt; [37] scales_1.2.1                SummarizedExperiment_1.30.2</span></span>
<span><span class="co">#&gt; [39] cli_3.6.1                   rmarkdown_2.22             </span></span>
<span><span class="co">#&gt; [41] crayon_1.5.2                ragg_1.2.5                 </span></span>
<span><span class="co">#&gt; [43] generics_0.1.3              cachem_1.0.8               </span></span>
<span><span class="co">#&gt; [45] stringr_1.5.0               zlibbioc_1.46.0            </span></span>
<span><span class="co">#&gt; [47] parallel_4.3.0              BiocManager_1.30.21        </span></span>
<span><span class="co">#&gt; [49] XVector_0.40.0              matrixStats_1.0.0          </span></span>
<span><span class="co">#&gt; [51] vctrs_0.6.3                 Matrix_1.6-0               </span></span>
<span><span class="co">#&gt; [53] jsonlite_1.8.5              bookdown_0.34              </span></span>
<span><span class="co">#&gt; [55] IRanges_2.34.0              S4Vectors_0.38.1           </span></span>
<span><span class="co">#&gt; [57] systemfonts_1.0.4           jpeg_0.1-10                </span></span>
<span><span class="co">#&gt; [59] jquerylib_0.1.4             glue_1.6.2                 </span></span>
<span><span class="co">#&gt; [61] pkgdown_2.0.7               codetools_0.2-19           </span></span>
<span><span class="co">#&gt; [63] stringi_1.7.12              gtable_0.3.3               </span></span>
<span><span class="co">#&gt; [65] GenomeInfoDb_1.36.0         deldir_1.0-9               </span></span>
<span><span class="co">#&gt; [67] GenomicRanges_1.52.0        munsell_0.5.0              </span></span>
<span><span class="co">#&gt; [69] tibble_3.2.1                pillar_1.9.0               </span></span>
<span><span class="co">#&gt; [71] htmltools_0.5.5             GenomeInfoDbData_1.2.10    </span></span>
<span><span class="co">#&gt; [73] R6_2.5.1                    textshaping_0.3.6          </span></span>
<span><span class="co">#&gt; [75] Rdpack_2.4                  rprojroot_2.0.3            </span></span>
<span><span class="co">#&gt; [77] evaluate_0.21               Biobase_2.60.0             </span></span>
<span><span class="co">#&gt; [79] lattice_0.21-8              rbibutils_2.2.13           </span></span>
<span><span class="co">#&gt; [81] png_0.1-8                   Rsamtools_2.16.0           </span></span>
<span><span class="co">#&gt; [83] memoise_2.0.1               bslib_0.5.0                </span></span>
<span><span class="co">#&gt; [85] Rcpp_1.0.10                 gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [87] xfun_0.39                   fs_1.6.2                   </span></span>
<span><span class="co">#&gt; [89] MatrixGenerics_1.12.2       pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wenjie Sun, Anne-Marie Lyne, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
