<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analyzing Cellular DNA Barcode with CellBarcode • CellBarcode</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing Cellular DNA Barcode with CellBarcode">
<meta property="og:description" content="CellBarcode">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-NY9M0DHR4F"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-NY9M0DHR4F');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CellBarcode</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Barcode_in_10X_scRNASeq.html">Get barcode from 10X Genomics scRNASeq data</a>
    </li>
    <li>
      <a href="../articles/UMI_VDJ_Barcode.html">Analyzing Cellular DNA Barcode with CellBarcode</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analyzing Cellular DNA Barcode with
CellBarcode</h1>
                        <h4 data-toc-skip class="author">Wenjie Sun</h4>
                                    <h4 data-toc-skip class="author">Anne-Marie
Lyne</h4>
                        
      
      
      <div class="hidden name"><code>UMI_VDJ_Barcode.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://r-datatable.com" class="external-link">data.table</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CellBarcode</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="about-the-package">About the package<a class="anchor" aria-label="anchor" href="#about-the-package"></a>
</h3>
<p><strong>What’s this package used for?</strong></p>
<p>Cellular DNA barcoding (genetic lineage tracing) is a powerful tool
for lineage tracing and clonal tracking studies. This package provides a
toolbox for DNA barcode analysis, from extraction from fastq files to
barcode error correction and quantification.</p>
<p><strong>What types of barcode can this package handle?</strong></p>
<p>The package can handle all kinds of barcodes, as long as the barcodes
have a pattern which can be matched by a regular expression, and each
barcode is within a single sequencing read. It can handle barcodes with
flexible length, and barcodes with UMI (unique molecular
identifier).</p>
<p>This tool can also be used for the pre-processing part of amplicon
data analysis such as CRISPR gRNA screening, immune repertoire
sequencing and meta genome data.</p>
<p><strong>What can the package do?</strong></p>
<p>The package provides functions for 1). Sequence quality control and
filtering, 2). Barcode (and UMI) extraction from sequencing reads, 3).
Sample and barcode management with metadata, 4). Barcode filtering.</p>
</div>
<div class="section level3">
<h3 id="about-function-naming">About function naming<a class="anchor" aria-label="anchor" href="#about-function-naming"></a>
</h3>
<p>Most of the functions in this packages have names with
<code>bc_</code> as initiation. We hope it can facilitate the syntax
auto-complement function of IDE (integrated development toolkit) or
IDE-like tools such as RStudio, R-NVIM (in VIM), and ESS (in Emacs). By
typing <code>bc_</code> you can have a list of suggested functions, then
you can pick the function you need from the list.</p>
<p>TODO: the function brain-map</p>
</div>
<div class="section level3">
<h3 id="about-test-data-set">About test data set<a class="anchor" aria-label="anchor" href="#about-test-data-set"></a>
</h3>
<p>The test data set in this package can be accessed by</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mef_test_data"</span>, package<span class="op">=</span><span class="st">"CellBarcode"</span><span class="op">)</span></span></code></pre></div>
<p>The data are from Jos et. al (TODO: citation). There are 7 mouse
embryo fibroblast (MEF) lines with different DNA barcodes. The barcodes
are in vivo inducible VDJ barcodes (TODO: add citation when have). These
MEF lines were mixed in a ratio of 1:2:4:8:16:32:64.</p>
<table class="table">
<thead><tr class="header">
<th>sequence</th>
<th>clone size 2^x</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>AAGTCCAGTTCTACTATCGTAGCTACTA</td>
<td>1</td>
</tr>
<tr class="even">
<td>AAGTCCAGTATCGTTACGCTACTA</td>
<td>2</td>
</tr>
<tr class="odd">
<td>AAGTCCAGTCTACTATCGTTACGACAGCTACTA</td>
<td>3</td>
</tr>
<tr class="even">
<td>AAGTCCAGTTCTACTATCGTTACGAGCTACTA</td>
<td>4</td>
</tr>
<tr class="odd">
<td>AAGTCCATCGTAGCTACTA</td>
<td>5</td>
</tr>
<tr class="even">
<td>AAGTCCAGTACTGTAGCTACTA</td>
<td>6</td>
</tr>
<tr class="odd">
<td>AAGTCCAGTACTATCGTACTA</td>
<td>7</td>
</tr>
</tbody>
</table>
<p>Then 5 pools of 196 to 50000 cells were prepared from the MEF lines
mixture. For each pool 2 technical replicates (NGS libraries) were
prepared and sequenced, finally resulting in 10 samples.</p>
<table class="table">
<thead><tr class="header">
<th>sample name</th>
<th>cell number</th>
<th>replication</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>195_mixa</td>
<td>195</td>
<td>mixa</td>
</tr>
<tr class="even">
<td>195_mixb</td>
<td>195</td>
<td>mixb</td>
</tr>
<tr class="odd">
<td>781_mixa</td>
<td>781</td>
<td>mixa</td>
</tr>
<tr class="even">
<td>781_mixb</td>
<td>781</td>
<td>mixb</td>
</tr>
<tr class="odd">
<td>3125_mixa</td>
<td>3125</td>
<td>mixa</td>
</tr>
<tr class="even">
<td>3125_mixb</td>
<td>3125</td>
<td>mixb</td>
</tr>
<tr class="odd">
<td>12500_mixa</td>
<td>12500</td>
<td>mixa</td>
</tr>
<tr class="even">
<td>12500_mixb</td>
<td>12500</td>
<td>mixb</td>
</tr>
<tr class="odd">
<td>50000_mixa</td>
<td>50000</td>
<td>mixa</td>
</tr>
<tr class="even">
<td>50000_mixb</td>
<td>50000</td>
<td>mixb</td>
</tr>
</tbody>
</table>
<p>The original FASTQ files are relatively large, so only 2000 reads for
each sample have been randomly sampled as a test set here.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mef_test_data"</span>, package <span class="op">=</span> <span class="st">"CellBarcode"</span><span class="op">)</span></span>
<span><span class="va">fq_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">example_data</span>, <span class="st">"fastq.gz"</span>, full<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># prepare metadata for the samples</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_split.html" class="external-link">str_split_fixed</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">fq_files</span><span class="op">)</span>, <span class="st">"_"</span>, <span class="fl">10</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span>
<span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">metadata</span>, <span class="fl">1</span>, <span class="va">paste</span>, collapse <span class="op">=</span> <span class="st">"_"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cell_number"</span>, <span class="st">"replication"</span><span class="op">)</span></span>
<span><span class="co"># metadata should has the row names consistent to the sample names</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span> <span class="op">=</span> <span class="va">sample_name</span></span>
<span><span class="va">metadata</span></span>
<span><span class="co">#&gt;            cell_number replication</span></span>
<span><span class="co">#&gt; 50000_mixa       50000        mixa</span></span>
<span><span class="co">#&gt; 195_mixb           195        mixb</span></span>
<span><span class="co">#&gt; 50000_mixb       50000        mixb</span></span>
<span><span class="co">#&gt; 12500_mixa       12500        mixa</span></span>
<span><span class="co">#&gt; 12500_mixb       12500        mixb</span></span>
<span><span class="co">#&gt; 3125_mixa         3125        mixa</span></span>
<span><span class="co">#&gt; 3125_mixb         3125        mixb</span></span>
<span><span class="co">#&gt; 781_mixa           781        mixa</span></span>
<span><span class="co">#&gt; 781_mixb           781        mixb</span></span>
<span><span class="co">#&gt; 195_mixa           195        mixa</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Install from Bioconductor.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"CellBarcode"</span><span class="op">)</span></span></code></pre></div>
<p>Install the development version from Github.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"wenjie1991/CellBarcode"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-basic-workflow">A basic workflow<a class="anchor" aria-label="anchor" href="#a-basic-workflow"></a>
</h2>
<p>Here is an example of a basic workflow:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("stringr")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CellBarcode</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The example data is the mix of MEF lines with known barcodes</span></span>
<span><span class="co"># 2000 reads for each file have been sampled for this test dataset</span></span>
<span><span class="co"># extract UMI barcode with regular expression</span></span>
<span><span class="va">bc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_extract.html">bc_extract</a></span><span class="op">(</span></span>
<span>  <span class="va">fq_files</span>,  <span class="co"># fastq file</span></span>
<span>  pattern <span class="op">=</span> <span class="st">"([ACGT]{12})CTCGAGGTCATCGAAGTATC([ACGT]+)CCGTAGCAAGCTCGAGAGTAGACCTACT"</span>, </span>
<span>  pattern_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMI"</span> <span class="op">=</span> <span class="fl">1</span>, <span class="st">"barcode"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>,</span>
<span>  sample_name <span class="op">=</span> <span class="va">sample_name</span>,</span>
<span>  metadata <span class="op">=</span> <span class="va">metadata</span></span>
<span><span class="op">)</span></span>
<span><span class="va">bc_obj</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; cell_number  replication  raw_read_count  barcode_read_count</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 10 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 1310 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixb there are: 1318 Tags</span></span>
<span><span class="co">#&gt;     In sample $50000_mixb there are: 1385 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 1321 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixb there are: 1361 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 1287 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixb there are: 1297 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 1295 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 1303 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 1343 Tags</span></span>
<span></span>
<span><span class="co"># sample subset operation, select technical repeats 'mixa'</span></span>
<span><span class="va">bc_sub</span> <span class="op">=</span> <span class="fu"><a href="../reference/bc_subset.html">bc_subset</a></span><span class="op">(</span><span class="va">bc_obj</span>, sample<span class="op">=</span><span class="va">replication</span> <span class="op">==</span> <span class="st">"mixa"</span><span class="op">)</span></span>
<span><span class="va">bc_sub</span> </span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; cell_number  replication  raw_read_count  barcode_read_count</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 5 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 1310 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 1321 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 1287 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 1295 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 1343 Tags</span></span>
<span></span>
<span><span class="co"># filter the barcode, UMI barcode amplicon &gt;= 2 &amp; UMI counts &gt;= 2</span></span>
<span><span class="va">bc_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_umi.html">bc_cure_umi</a></span><span class="op">(</span><span class="va">bc_sub</span>, depth <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span>depth <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select barcodes with a white list</span></span>
<span><span class="fu"><a href="../reference/bc_2df.html">bc_2df</a></span><span class="op">(</span><span class="va">bc_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name                      barcode_seq count</span></span>
<span><span class="co">#&gt; 1   50000_mixa         AAGTCCAGTATCGTTACGCTACTA    10</span></span>
<span><span class="co">#&gt; 2   50000_mixa           AAGTCCAGTACTGTAGCTACTA    11</span></span>
<span><span class="co">#&gt; 3   50000_mixa              AAGTCCATCGTAGCTACTA     3</span></span>
<span><span class="co">#&gt; 4   12500_mixa           AAGTCCAGTACTGTAGCTACTA    11</span></span>
<span><span class="co">#&gt; 5   12500_mixa         AAGTCCAGTATCGTTACGCTACTA    20</span></span>
<span><span class="co">#&gt; 6   12500_mixa              AAGTCCATCGTAGCTACTA     4</span></span>
<span><span class="co">#&gt; 7   12500_mixa AAGTCCAGTTCTACTATCGTTACGAGCTACTA     3</span></span>
<span><span class="co">#&gt; 8    3125_mixa              AAGTCCATCGTAGCTACTA     7</span></span>
<span><span class="co">#&gt; 9    3125_mixa         AAGTCCAGTATCGTTACGCTACTA    17</span></span>
<span><span class="co">#&gt; 10   3125_mixa AAGTCCAGTTCTACTATCGTTACGAGCTACTA     7</span></span>
<span><span class="co">#&gt; 11   3125_mixa           AAGTCCAGTACTGTAGCTACTA     9</span></span>
<span><span class="co">#&gt; 12    781_mixa         AAGTCCAGTATCGTTACGCTACTA     7</span></span>
<span><span class="co">#&gt; 13    781_mixa           AAGTCCAGTACTGTAGCTACTA     9</span></span>
<span><span class="co">#&gt; 14    781_mixa              AAGTCCATCGTAGCTACTA     2</span></span>
<span><span class="co">#&gt; 15    195_mixa              AAGTCCATCGTAGCTACTA     9</span></span>
<span><span class="co">#&gt; 16    195_mixa           AAGTCCAGTACTGTAGCTACTA    11</span></span>
<span><span class="co">#&gt; 17    195_mixa         AAGTCCAGTATCGTTACGCTACTA    12</span></span>
<span><span class="co">#&gt; 18    195_mixa AAGTCCAGTTCTACTATCGTTACGAGCTACTA     4</span></span>
<span><span class="co">#&gt; 19    195_mixa            AAGTCCAGTACTATCGTACTA     2</span></span>
<span><span class="va">bc_sub</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AAGTCCAGTACTATCGTACTA"</span>, <span class="st">"AAGTCCAGTACTGTAGCTACTA"</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 5 field(s) available:</span></span>
<span><span class="co">#&gt; cell_number  replication  raw_read_count  barcode_read_count  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 5 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 434 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 490 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 455 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 524 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 484 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 5 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 1 barcodes</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 1 barcodes</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 1 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 1 barcodes</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 2 barcodes</span></span>
<span></span>
<span><span class="co"># export the barcode counts to data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_2df.html">bc_2df</a></span><span class="op">(</span><span class="va">bc_sub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;   sample_name              barcode_seq count</span></span>
<span><span class="co">#&gt; 1  50000_mixa AAGTCCAGTATCGTTACGCTACTA    10</span></span>
<span><span class="co">#&gt; 2  50000_mixa   AAGTCCAGTACTGTAGCTACTA    11</span></span>
<span><span class="co">#&gt; 3  50000_mixa      AAGTCCATCGTAGCTACTA     3</span></span>
<span><span class="co">#&gt; 4  12500_mixa   AAGTCCAGTACTGTAGCTACTA    11</span></span>
<span><span class="co">#&gt; 5  12500_mixa AAGTCCAGTATCGTTACGCTACTA    20</span></span>
<span><span class="co">#&gt; 6  12500_mixa      AAGTCCATCGTAGCTACTA     4</span></span>
<span></span>
<span><span class="co"># export the barcode counts to matrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_2df.html">bc_2matrix</a></span><span class="op">(</span><span class="va">bc_sub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                  X12500_mixa X195_mixa X3125_mixa X50000_mixa</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTATCGTACTA                      0         2          0           0</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTGTAGCTACTA                    11        11          9          11</span></span>
<span><span class="co">#&gt; AAGTCCAGTATCGTTACGCTACTA                  20        12         17          10</span></span>
<span><span class="co">#&gt; AAGTCCAGTTCTACTATCGTTACGAGCTACTA           3         4          7           0</span></span>
<span><span class="co">#&gt; AAGTCCATCGTAGCTACTA                        4         9          7           3</span></span>
<span><span class="co">#&gt;                                  X781_mixa</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTATCGTACTA                    0</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTGTAGCTACTA                   9</span></span>
<span><span class="co">#&gt; AAGTCCAGTATCGTTACGCTACTA                 7</span></span>
<span><span class="co">#&gt; AAGTCCAGTTCTACTATCGTTACGAGCTACTA         0</span></span>
<span><span class="co">#&gt; AAGTCCATCGTAGCTACTA                      2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sequence-quality-control">Sequence quality control<a class="anchor" aria-label="anchor" href="#sequence-quality-control"></a>
</h2>
<div class="section level3">
<h3 id="evaluation">Evaluation<a class="anchor" aria-label="anchor" href="#evaluation"></a>
</h3>
<p>In a full analysis starting from fastq files, the first step is to
check the seqencing quality and filter as required. The
<code>bc_seq_qc</code> function is for checking the sequencing quality.
If multiple samples are input the output is a <code>BarcodeQcSet</code>
object, otherwise a <code>BarcodeQC</code> object will be returned. In
addition, <code>bc_seq_qc</code> also can handle the
<code>ShortReadQ</code>, <code>DNAStringSet</code> and other data
types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_noFilter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_seq_qc.html">bc_seq_qc</a></span><span class="op">(</span><span class="va">fq_files</span><span class="op">)</span></span>
<span><span class="va">qc_noFilter</span></span>
<span><span class="co">#&gt; The sequence QC set, use `[]` to select sample:</span></span>
<span><span class="co">#&gt;      5290_1_BCM_50000_mef_mixa_GTTCTCC_S2_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_10_BCM_195_mef_mixb_GTCATTG_S11_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_2_BCM_50000_mef_mixb_GATGTGT_S5_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_3_BCM_12500_mef_mixa_TGCCTTG_S4_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_4_BCM_12500_mef_mixb_TAACTGC_S8_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_5_BCM_3125_mef_mixa_GCTTCCA_S9_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_6_BCM_3125_mef_mixb_TGTGAGT_S7_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_7_BCM_781_mef_mixa_CCTTACC_S12_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_8_BCM_781_mef_mixb_CGTATCC_S13_R1_001.fastq.gz</span></span>
<span><span class="co">#&gt;      5290_9_BCM_195_mef_mixa_GTACTGT_S14_R1_001.fastq.gz</span></span>
<span><span class="fu"><a href="../reference/bc_names.html">bc_names</a></span><span class="op">(</span><span class="va">qc_noFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "5290_1_BCM_50000_mef_mixa_GTTCTCC_S2_R1_001.fastq.gz"</span></span>
<span><span class="co">#&gt;  [2] "5290_10_BCM_195_mef_mixb_GTCATTG_S11_R1_001.fastq.gz"</span></span>
<span><span class="co">#&gt;  [3] "5290_2_BCM_50000_mef_mixb_GATGTGT_S5_R1_001.fastq.gz"</span></span>
<span><span class="co">#&gt;  [4] "5290_3_BCM_12500_mef_mixa_TGCCTTG_S4_R1_001.fastq.gz"</span></span>
<span><span class="co">#&gt;  [5] "5290_4_BCM_12500_mef_mixb_TAACTGC_S8_R1_001.fastq.gz"</span></span>
<span><span class="co">#&gt;  [6] "5290_5_BCM_3125_mef_mixa_GCTTCCA_S9_R1_001.fastq.gz" </span></span>
<span><span class="co">#&gt;  [7] "5290_6_BCM_3125_mef_mixb_TGTGAGT_S7_R1_001.fastq.gz" </span></span>
<span><span class="co">#&gt;  [8] "5290_7_BCM_781_mef_mixa_CCTTACC_S12_R1_001.fastq.gz" </span></span>
<span><span class="co">#&gt;  [9] "5290_8_BCM_781_mef_mixb_CGTATCC_S13_R1_001.fastq.gz" </span></span>
<span><span class="co">#&gt; [10] "5290_9_BCM_195_mef_mixa_GTACTGT_S14_R1_001.fastq.gz"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">qc_noFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "BarcodeQcSet"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "CellBarcode"</span></span></code></pre></div>
<p>The <code>bc_plot_seqQc</code> function can be invoked with a
<code>BarcodeQcSet</code> as argument, and the output is a QC summary
with two panels. The first shows the ratio of ATCG bases for each
sequencing cycle with one sample per row; this allows the user to, for
example, identify constant or random parts of the sequencing read. The
second figure shows the average sequencing quality index of each cycle
(base).</p>
<p>For the test set, the first 12 bases are UMI, which are random. This
is followed by the constant region of the barcode (the PCR primer
selects reads with this sequence), and here we observe a specific base
for each cycle across all the samples.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_seq_qc.html">bc_plot_seqQc</a></span><span class="op">(</span><span class="va">qc_noFilter</span><span class="op">)</span> </span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/qc_figure_set1-1.png" width="700"></p>
<p>We can also plot one of the <code>BarcodeQc</code> in the
<code>BarcodeQcSet</code> object. In the output, there are three panels.
The top left one shows the reads depth distribution, the top right
figure shows the “ATCG” base ratio by each sequencing cycle, and the
last one shows the sequencing quality by sequencing cycle.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qc_noFilter</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="co">#&gt; Sequnece QC, summary:</span></span>
<span><span class="co">#&gt;      total_read: 2000</span></span>
<span><span class="co">#&gt;      p5_read_length: 100</span></span>
<span><span class="co">#&gt;      median_read_length: 100</span></span>
<span><span class="co">#&gt;      p95_read_length: 100</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">qc_noFilter</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "BarcodeQc"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "CellBarcode"</span></span>
<span><span class="fu"><a href="../reference/bc_seq_qc.html">bc_plot_seqQc</a></span><span class="op">(</span><span class="va">qc_noFilter</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span> </span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/qc_figure_single1-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h3>
<p><code>bc_seq_filter</code> reads in the sequence data and applies
filters, then returns a <code>ShortReadQ</code> object which contains
the filtered sequences.</p>
<p>The <code>bc_seq_filter</code> function can read fastq files, and it
can also handle sequencing reads in <code>ShortReadQ</code>,
<code>DNAStringSet</code> and <code>data.frame</code>.</p>
<p>The currently available filter parameters are: - min_average_quality:
average base sequencing quality across read. - min_read_length: minimum
number of bases per read. - N_threshold: maximum number of “N” bases in
sequence.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># TODO: output the filtering percentage</span></span>
<span><span class="co"># TODO: Trimming</span></span>
<span><span class="va">fq_filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_seq_filter.html">bc_seq_filter</a></span><span class="op">(</span></span>
<span>  <span class="va">fq_files</span>,</span>
<span>  min_average_quality <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  min_read_length <span class="op">=</span> <span class="fl">60</span>,</span>
<span>  sample_name <span class="op">=</span> <span class="va">sample_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fq_filter</span></span>
<span><span class="co">#&gt; $`50000_mixa`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1154 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`195_mixb`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1187 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`50000_mixb`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1234 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`12500_mixa`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1186 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`12500_mixb`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1238 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`3125_mixa`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1100 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`3125_mixb`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1146 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`781_mixa`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1154 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`781_mixb`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1164 reads; width: 100 cycles</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`195_mixa`</span></span>
<span><span class="co">#&gt; class: ShortReadQ</span></span>
<span><span class="co">#&gt; length: 1205 reads; width: 100 cycles</span></span>
<span><span class="fu"><a href="../reference/bc_seq_qc.html">bc_plot_seqQc</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_seq_qc.html">bc_seq_qc</a></span><span class="op">(</span><span class="va">fq_filter</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/filter_low_quality_seq-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="parse-reads">Parse reads<a class="anchor" aria-label="anchor" href="#parse-reads"></a>
</h2>
<p>One of the core applications of this package is parsing the sequences
to get the barcode (and UMI). Our package uses regular expressions to
identify barcodes (and UMI) from sequencing reads. This is how we tell
<code>bc_extract</code> the structure of the input sequences.</p>
<p>3 arguments are necessary for <code>bc_extract</code>, they are: - x:
the sequence data, it can be in fastq, <code>ShortReadQ</code>,
<code>DNAStringSet</code> or <code>data.frame</code> format. - pattern:
the sequence pattern regular expression. - pattern_type: pattern
description.</p>
<p>The <code>pattern</code> argument is the regular expression, it tells
the function where to find the barcode (or UMI). We capture the barcode
(or UMI) by <code>()</code> in the backbone. For the sequence captured
by <code>()</code>, the <code>pattern_type</code> argument tells which
is the UMI or the barcode. In the example</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pattern</span> <span class="op">&lt;-</span> <span class="st">"([ACGA]{12})CTCGAGGTCATCGAAGTATC([ACGT]+)CCGTAGCAAGCTCGAGAGTAGACCTACT"</span></span>
<span><span class="va">pattern_type</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMI"</span> <span class="op">=</span> <span class="fl">1</span>, <span class="st">"barcode"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>The sequence starts with 12 base pairs of random sequence, which is
UMI. It is the first barcode captured by <code>()</code> in the
<code>pattern</code> argument, and corresponds to <code>UMI = 1</code>
in the <code>pattern_type</code> argument.</li>
<li>Then, there is a known constant sequence:
<code>CTCGAGGTCATCGAAGTATC</code>.</li>
<li>Following the constant region, there is flexible length random
sequence. This is the barcode which is trapped by second
<code>()</code>, and it is defined by <code>barcode = 2</code> in the
<code>pattern_type</code> argument.</li>
<li>At the end of the sequence, there is another constant sequence
<code>CCGTAGCAAGCTCGAGAGTAGACCTACT</code>.</li>
</ol>
<p>In the regular expression, the UMI pattern is retrieved with
<code>[ACGT]{12}</code>. The <code>[ACGT]</code> means to match “A”,
“C”, “G” or “T”, and the <code>{12}</code> means match 12
<code>[ACGT]</code>. In the barcode pattern <code>[ACGT]+</code>, again
<code>[ACGT]</code> means match “A”, “C”, “G” or “T” and the
<code>+</code> says to match at least one <code>[ACGT]</code>.</p>
<p>The <code>bc_extract</code> function is used to extract the
barcode(s) from the sequences. It returns a <code>BarcodeObj</code>
object if the input is a list or a vector of Fastq files. The
<code>BarcodeObj</code> created by <code>bc_extract</code> is a R S4
class with three slots: <code>messyBc</code>, <code>metadata</code> and
<code>cleanBc</code> (which is NULL in the <code>bc_extract</code>
output). They can be accessed by <code>@</code> operator or
corresponding accesors: - <code>bc_messyBc</code>: return the
<code>messyBc</code> slot. - <code>bc_cleanBc</code>: return the
<code>cleanBc</code> slot. - <code>bc_meta</code>: return the
<code>metadata</code> slot.</p>
<p><code>messyBc</code> is a list, where each element is a
<code>data.table</code> corresponding to the successive samples. Each
<code>data.table</code> has 3 columns:</p>
<ol style="list-style-type: decimal">
<li>umi_seq (optional): UMI sequence, applicable when there is a UMI in
<code>pattern</code> and <code>pattern_type</code> argument.</li>
<li>barcode_seq: barcode sequence.</li>
<li>count: the count of the full read sequence.</li>
</ol>
<p><strong>Attention</strong>: In the <code>data.table</code>,
<code>barcode_seq</code> value may be not unique, as two different full
read sequences can contain the same barcode sequence, due to the UMI or
mutations in the constant region.</p>
<p>If the input to <code>bc_extract</code> is just a sample, the output
is a single <code>data.frame</code> with the 3 columns 1).
<code>umi_seq</code>, 2). <code>barcode_seq</code> and 3).
<code>count</code>, as described above.</p>
<p>The sequence in <code>match_seq</code> is a contiguous segment of the
full read given in <code>reads_seq</code>. The <code>umi_seq</code> and
<code>barcode_seq</code> are contiguous segments of
<code>match_seq</code>. Take note that, the <code>reads_seq</code> is
the unique id for each row. The <code>match_seq</code>,
<code>umi_seq</code> or <code>barcode_seq</code> can be duplicated, due
to the potential variation in the region outside of
<code>match_seq</code>. Please keep this in mind when you use data in
<code>$messyBc</code> to perform the analysis.</p>
<div class="section level3">
<h3 id="sequencing-without-umi">Sequencing without UMI<a class="anchor" aria-label="anchor" href="#sequencing-without-umi"></a>
</h3>
<p>In the following example, only a barcode is extracted.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pattern</span> <span class="op">&lt;-</span> <span class="st">"CTCGAGGTCATCGAAGTATC([ACGT]+)CCGTAGCAAGCTCGAGAGTAGACCTACT"</span></span>
<span><span class="va">bc_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_extract.html">bc_extract</a></span><span class="op">(</span></span>
<span>  <span class="va">fq_filter</span>,</span>
<span>  sample_name <span class="op">=</span> <span class="va">sample_name</span>,</span>
<span>  pattern <span class="op">=</span> <span class="va">pattern</span>,</span>
<span>  pattern_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>barcode <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bc_obj</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 2 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 10 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 39 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixb there are: 31 Tags</span></span>
<span><span class="co">#&gt;     In sample $50000_mixb there are: 49 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 37 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixb there are: 44 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 36 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixb there are: 44 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 34 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 29 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 34 Tags</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_messyBc.html">bc_messyBc</a></span><span class="op">(</span><span class="va">bc_obj</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "barcode_seq" "count"</span></span></code></pre></div>
<p>Here the regular expression matches a constant sequence at the
beginning and the end and the barcode in <code>()</code> matches at
least one of any character.</p>
</div>
<div class="section level3">
<h3 id="sequencing-with-umi">Sequencing with UMI<a class="anchor" aria-label="anchor" href="#sequencing-with-umi"></a>
</h3>
<p>In the following example, both UMI and barcode are extracted. The
regular expression is explained above.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pattern</span> <span class="op">&lt;-</span> <span class="st">"([ACGA]{12})CTCGAGGTCATCGAAGTATC([ACGT]+)CCGTAGCAAGCTCGAGAGTAGACCTACT"</span></span>
<span><span class="va">bc_obj_umi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_extract.html">bc_extract</a></span><span class="op">(</span></span>
<span>  <span class="va">fq_filter</span>,</span>
<span>  sample_name <span class="op">=</span> <span class="va">sample_name</span>,</span>
<span>  pattern <span class="op">=</span> <span class="va">pattern</span>,</span>
<span>  maxLDist <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  pattern_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>UMI <span class="op">=</span> <span class="fl">1</span>, barcode <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">bc_obj_umi</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "BarcodeObj"</span></span>
<span><span class="co">#&gt; attr(,"package")</span></span>
<span><span class="co">#&gt; [1] "CellBarcode"</span></span>
<span><span class="va">bc_obj_umi</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 2 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 10 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $50000_mixa there are: 142 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixb there are: 127 Tags</span></span>
<span><span class="co">#&gt;     In sample $50000_mixb there are: 143 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixa there are: 106 Tags</span></span>
<span><span class="co">#&gt;     In sample $12500_mixb there are: 134 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixa there are: 106 Tags</span></span>
<span><span class="co">#&gt;     In sample $3125_mixb there are: 120 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt;     In sample $195_mixa there are: 154 Tags</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="metadata-updated">Metadata updated<a class="anchor" aria-label="anchor" href="#metadata-updated"></a>
</h3>
<p><code>bc_extract</code> added two columns named “row_read_count” and
“barcode_read_count” to the metadata slot of the returned BarcodeObj
object.</p>
<p><strong>row_read_count</strong>: Total raw reads number of each
sample. <strong>barcode_read_count</strong>: The number of reads that
contain the barcodes.</p>
<p>You can use the ratio of <code>barcode_read_count</code> versus
<code>raw_read_count</code> to check the successfulness of the
sequencing or correctness of the pattern given to the
<code>bc_extract</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select two samples from bc_obj_umi</span></span>
<span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="va">bc_obj_umi</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"781_mixa"</span>, <span class="st">"781_mixb"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co"># get the metadata matrix</span></span>
<span><span class="op">(</span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_meta.html">bc_meta</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;          raw_read_count barcode_read_count</span></span>
<span><span class="co">#&gt; 781_mixa           1154                134</span></span>
<span><span class="co">#&gt; 781_mixb           1164                112</span></span>
<span><span class="co"># use the row name of the metadata, which contains the sample names</span></span>
<span><span class="va">d</span><span class="op">$</span><span class="va">sample_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span></span>
<span><span class="va">d</span><span class="op">$</span><span class="va">barcode_read_count</span> <span class="op">/</span> <span class="va">d</span><span class="op">$</span><span class="va">raw_read_count</span></span>
<span><span class="co">#&gt; [1] 0.11611785 0.09621993</span></span>
<span><span class="co"># visualize</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">sample_name</span>, y<span class="op">=</span><span class="va">barcode_read_count</span> <span class="op">/</span> <span class="va">raw_read_count</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/unnamed-chunk-4-1.png" width="480"></p>
</div>
</div>
<div class="section level2">
<h2 id="data-management">Data management<a class="anchor" aria-label="anchor" href="#data-management"></a>
</h2>
<p>Besides, we provide operators to handle the barcodes and samples in
<code>BarcodeObj</code> object. You can easily select one or several
samples by their names, indices or metadata.</p>
<p>Select slot by accesors:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Access messyBc slot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_messyBc.html">bc_messyBc</a></span><span class="op">(</span><span class="va">bc_obj_umi</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;        umi_seq              barcode_seq count</span></span>
<span><span class="co">#&gt; 1 GGACAGAGCGGA AAGTCCAGTATCGTTACGCTACTA     2</span></span>
<span><span class="co">#&gt; 2 GGCCGGCGGGGC   AAGTCCAGTACTGTAGCTACTA     2</span></span>
<span><span class="co"># return a data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_messyBc.html">bc_messyBc</a></span><span class="op">(</span><span class="va">bc_obj_umi</span>, isList<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name      umi_seq              barcode_seq count</span></span>
<span><span class="co">#&gt; 1:  50000_mixa GGACAGAGCGGA AAGTCCAGTATCGTTACGCTACTA     2</span></span>
<span><span class="co">#&gt; 2:  50000_mixa GGCCGGCGGGGC   AAGTCCAGTACTGTAGCTACTA     2</span></span>
<span></span>
<span><span class="co"># Access cleanBc slot</span></span>
<span><span class="co"># return a data.frame</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/bc_cleanBc.html">bc_cleanBc</a></span><span class="op">(</span><span class="va">bc_obj_umi</span>, isList<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>, n<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_nmae      umi_seq              barcode_seq count</span></span>
<span><span class="co">#&gt; 1:  50000_mixa GGACAGAGCGGA AAGTCCAGTATCGTTACGCTACTA     2</span></span>
<span><span class="co">#&gt; 2:  50000_mixa GGCCGGCGGGGC   AAGTCCAGTACTGTAGCTACTA     2</span></span></code></pre></div>
<p>Select sample by sample names</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="va">bc_obj_umi</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"781_mixa"</span>, <span class="st">"781_mixb"</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="../reference/bc_names.html">bc_names</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "781_mixa" "781_mixb"</span></span></code></pre></div>
<p>Set metadata</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_meta.html">bc_meta</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span><span class="op">$</span><span class="va">rep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"a"</span>, <span class="st">"b"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/bc_meta.html">bc_meta</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt;          raw_read_count barcode_read_count rep</span></span>
<span><span class="co">#&gt; 781_mixa           1154                134   a</span></span>
<span><span class="co">#&gt; 781_mixb           1164                112   b</span></span></code></pre></div>
<p>Select sample by metadata</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_subset.html">bc_subset</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span>, sample <span class="op">=</span> <span class="va">rep</span> <span class="op">==</span> <span class="st">"a"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 3 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 1 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="barcode-filtering">Barcode filtering<a class="anchor" aria-label="anchor" href="#barcode-filtering"></a>
</h2>
<p>Most of the times, it needs PCR and NGS to read out the cellular
barcode sequences. <code>bc_extract</code> will output all barcodes
found in the sequences. Some of the identified barcodes may contain PCR
or sequencing errors.</p>
<p>The potential errors derived from PCR and NGS lead to spurious
barcodes that not existed in biological samples. The spurious barcodes
are more likely to be less abundant comparing to corresponding “mother”
barcodes they derived from.</p>
<p>As UMI can be used to label a DNA molecular, one UMI labeled barcode
molecular becomes multiple copies by PCR. Thus all the sequences derived
from the template sequence, including original template sequence and
mutant ones, are marked by UMI for having the same UMI. The original
template sequence is likely having more reads comparing to the spurious
one derived from PCR or sequencing mutation, as errors happens with low
probability. Also, a barcode sequence is less likely to be spurious one
when it relates to several UMIs.</p>
<p>We created the <code>bc_cure_*</code> functions to perform filtering
for removing the potential spurious barcodes. The <code>bc_cure_*</code>
functions create or update the <code>cleanBc</code> slot in
<code>BarcodeObj</code>. The <code>cleanBc</code> slot contains 2
columns - barcode_seq: barcode sequence. - counts: reads count, or UMI
count in the case that the cleanBc was created by
<code>bc_cure_umi</code>.</p>
<p><strong>Important</strong>: The <code>createBc</code> slot, the
barcode_seq is not duplicated in each sample.</p>
<p>In the <code>bc_cure_*</code> function family, there are
<code>bc_cure_depth</code>, <code>bc_cure_umi</code> and
<code>bc_cure_cluster</code>.</p>
<div class="section level3">
<h3 id="filter-umi-barcode-tag">Filter UMI-barcode tag<a class="anchor" aria-label="anchor" href="#filter-umi-barcode-tag"></a>
</h3>
<p>In the case when the UMI is applied, the template sequence is marked
by UMI, and we use “UMI-barcode tag” to denote a combination of a UMI
and a barcode. The UMI-barcode tag with few reads are likely deriving
from PCR or sequence errors. <code>bc_cure_umi</code> carries out the
filtering based on the UMI-barcode tag read count from the
<code>messyBc</code> slot in BarcodeObj object, and returns a updated
BarcodeObj object with a <code>cleanBc</code> slot containing the
barcodes passing the filtering.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter the barcodes with UMI-barcode tag &gt;= 1, </span></span>
<span><span class="co"># and treat UMI as absolute unique and do "fish"</span></span>
<span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_umi.html">bc_cure_umi</a></span><span class="op">(</span></span>
<span>    <span class="va">bc_obj_umi_sub</span>, depth <span class="op">=</span> <span class="fl">1</span>, </span>
<span>    isUniqueUMI <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>    doFish <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">bc_obj_umi_sub</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 3 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 9 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 8 barcodes</span></span></code></pre></div>
<p>The available arguments of <code>bc_cure_umi</code> are:</p>
<ul>
<li>depth: minimum read count required for a UMI.</li>
<li>doFish: if true, for barcodes with UMI read depth above the
threshold, “fish” for identical barcodes with UMI read depth below the
threshold. The consequence of “doFish” will not increase the number of
identified barcodes, but the UMI counts will increase due to including
the low depth UMI barcodes.</li>
<li>isUniqueUMI: one UMI sequence may be linked to several barcodes. Do
you believe the UMI is absolutely unique? If yes, we treat the UMI as
absolutely unique makers. Thus the most abundant barcode will be picked
for a UMI, and less abundant barcodes with the same UMI are
obsolete.</li>
</ul>
</div>
<div class="section level3">
<h3 id="filter-by-count">Filter by count<a class="anchor" aria-label="anchor" href="#filter-by-count"></a>
</h3>
<p><code>bc_cure_depth</code> performs filtering by reads/UMI count. It
can filter the raw barcodes in the <code>messyBc</code> and create a
<code>cleanBc</code> slot , or update the <code>cleanBc</code> when the
argument <code>isUpdate</code> is TRUE. You should set this argument to
<code>TRUE</code>, when you want apply the filtering on the UMI count
with the <code>bc_cure_umi</code> output. In this case,
<code>bc_cure_depth</code> will update the <code>cleanBc</code> slot
created by <code>bc_cure_umi</code>.</p>
<p>The function has two arguments:</p>
<ul>
<li>depth: sequence/UMI count threshold, it can be a numeric number or a
numeric vector, in the later case, each number corresponds to a sample
in the BarcodeObj object.</li>
<li>isUpdate: if true (default) the bc_cure will preferentially perform
filtering on the <code>cleanBc</code> slot and update it, otherwise the
<code>messyBc</code> will be used as input.</li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply the barcode sequence depth with depth &gt;= 3</span></span>
<span><span class="co"># With isUpdate = FLASE, the data in `messyBc` slot of bc_obj_umi_sub</span></span>
<span><span class="co">#   will be used for depth filtering. The UMI information will be discarded, </span></span>
<span><span class="co">#   the identical barcodes in different UMI-barcode tags are merged before</span></span>
<span><span class="co">#   performing the sequence depth filtering.</span></span>
<span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span>, depth <span class="op">=</span> <span class="fl">3</span>, isUpdate <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">bc_obj_umi_sub</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 5 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 4 barcodes</span></span>
<span></span>
<span><span class="co"># Apply the UMI count filter, keep barcode &gt;= 3 UMI</span></span>
<span><span class="co"># The `bc_cure_umi` function applies the filtering on the UMI-barcode tags,</span></span>
<span><span class="co">#   and create a `cleanBc` slot in the return BarcodeObj object. Then, </span></span>
<span><span class="co">#   the `bc_cure_depth` with `isUpdate` argument TRUE will apply the filtering</span></span>
<span><span class="co">#   on the UMI counts in `cleanBc` and updated the `cleanBc`.</span></span>
<span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_umi.html">bc_cure_umi</a></span><span class="op">(</span></span>
<span>    <span class="va">bc_obj_umi_sub</span>, depth <span class="op">=</span> <span class="fl">1</span>, </span>
<span>    isUniqueUMI <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>    doFish <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">bc_obj_umi_sub</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 9 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 8 barcodes</span></span>
<span><span class="va">bc_obj_umi_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span>, depth <span class="op">=</span> <span class="fl">3</span>, isUpdate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">bc_obj_umi_sub</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 5 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 4 barcodes</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cluster-barcode-by-sequence-similarity">Cluster barcode by sequence similarity<a class="anchor" aria-label="anchor" href="#cluster-barcode-by-sequence-similarity"></a>
</h3>
<p>The sequences with more reads have more chance to be the original
templates. In contrast，the sequences with few reads are more likely
derived from mutations of the most abundant sequence. Thus, the spurious
sequence might be identified by comparing the most abundant sequence to
the least one. If they are similar, the least abundant sequence will be
removed.</p>
<p><code>bc_cure_cluster</code> performs the clustering to remove the
barcodes with insufficient depth (or UMI counts) comparing to most
abundant ones with similarity, it needs the <code>cleanBc</code> slot
and will update it.</p>
<p>To control the clustering methods and threshold for merging you need
the following arguments:</p>
<ul>
<li>dist_thresh: a single integer or vector of integers with the length
of sample number, specifying the editing distance threshold of merging
two similar barcode sequences. If the input is a vector, each value in
vector is for one sample according to the sample order in BarcodeObj
object.</li>
<li>dist_method: A character string, specifying the distance algorithm
for evaluating barcodes similarity. It can be “hamm” for Hamming
distance or “leven” for Levenshtein distance.</li>
<li>cluster_method: A character string specifying the algorithm used to
perform the clustering merging of barcodes. Currently only “greedy” is
available, in this case, the least abundant barcode is removed.</li>
<li>param count_threshold: An integer, read depth threshold to consider
a barcode as a true barcode, when when a barcode with count higher than
this threshold it will not be removed.</li>
<li>dist_costs: A list, the cost of the events when calculating distance
between two barcode sequences, applicable when Levenshtein distance is
applied. The names of vector have to be “insert”, “delete” and
“replace”, specifying the weight of insertion, deletion, replacement
events respectively. The default cost for each event is 1.</li>
</ul>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Do the clustering and merging the least abundant barcodes to the similar</span></span>
<span><span class="co"># abundant ones</span></span>
<span><span class="fu"><a href="../reference/bc_cure_cluster.html">bc_cure_cluster</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 4 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  rep  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 131 Tags</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 112 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixa there are: 5 barcodes</span></span>
<span><span class="co">#&gt;     In sample $781_mixb there are: 4 barcodes</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="barcode-count-distribution">Barcode count distribution<a class="anchor" aria-label="anchor" href="#barcode-count-distribution"></a>
</h2>
<p>We provides <code>bc_plot_single</code>, <code>bc_plot_mutual</code>
and <code>bc_plot_pair</code> functions for helping exploring the
barcode count distribution for single sample or between two samples.</p>
<div class="section level3">
<h3 id="single-sample">Single sample<a class="anchor" aria-label="anchor" href="#single-sample"></a>
</h3>
<p><code>bc_plot_single</code> can be used for exploring barcode count
distribution sample wise. It uses the <code>cleanBc</code> slot in the
BarcodeObj bc_obj_umi_sub.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_plot_single.html">bc_plot_single</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/unnamed-chunk-9-1.png" width="480"></p>
<p>The <code>bc_plot_single</code> function provides arguments for
showing the potential cutoff point and highlighting specific
barcodes.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># re-do the filtering using depth threshold 0 to include all barcodes.</span></span>
<span><span class="va">bc_obj_umi_sub_neo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span>, depth<span class="op">=</span><span class="fl">0</span>, isUpdate<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># you can use the count_marks argument to display the cutoff points in the figure</span></span>
<span><span class="co"># and the highlight argument to highlight specific barcodes.</span></span>
<span><span class="fu"><a href="../reference/bc_plot_single.html">bc_plot_single</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub_neo</span>, count_marks<span class="op">=</span><span class="fl">10</span>, </span>
<span>    highlight<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AAGTCCAGTACTATCGTACTA"</span>, <span class="st">"AAGTCCAGTACTGTAGCTACTA"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/unnamed-chunk-10-1.png" width="480"></p>
</div>
<div class="section level3">
<h3 id="pairwise">Pairwise<a class="anchor" aria-label="anchor" href="#pairwise"></a>
</h3>
<p><code>bc_plot_mutual</code> and <code>bc_plot_pair</code> are
designed for comparing the barcodes between two samples.</p>
<p>The <code>bc_plot_mutual</code> generates a scatter plot matrix which
contains all the pairwise sample combination in the provided BarcodeObj
object.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create a new BarcodeObj for following visualization</span></span>
<span><span class="co"># use depth as 0 to include all the barcodes.</span></span>
<span><span class="va">bc_obj_umi_neo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_umi</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, depth<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span><span class="co"># you can set the count_marks to display the cutoff point</span></span>
<span><span class="co"># and highlight specific barcodes dots by highlight</span></span>
<span><span class="fu"><a href="../reference/bc_plot_mutual.html">bc_plot_mutual</a></span><span class="op">(</span><span class="va">bc_obj_umi_neo</span>, count_marks<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span>, <span class="fl">40</span><span class="op">)</span>, </span>
<span>    highlight<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AAGTCCAGTACTATCGTACTA"</span>, <span class="st">"AAGTCCAGTACTGTAGCTACTA"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>And the <code>bc_plot_pair</code> only draws the scatter plot for the
given sample pairs.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create a new BarcodeObj for following visualization</span></span>
<span><span class="co"># use depth as 0 to include all the barcodes.</span></span>
<span><span class="va">bc_obj_umi_neo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_umi</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>, depth<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2d scatters plot with x axis of sample_x and y axis of sample_y</span></span>
<span><span class="co"># sample_x, and sample_y can be the sample name or sample index</span></span>
<span><span class="fu"><a href="../reference/bc_plot_pair.html">bc_plot_pair</a></span><span class="op">(</span></span>
<span>    <span class="va">bc_obj_umi_neo</span>, </span>
<span>    sample_x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"50000_mixa"</span><span class="op">)</span>,</span>
<span>    sample_y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"50000_mixb"</span>, <span class="st">"12500_mixa"</span>, <span class="st">"195_mixb"</span><span class="op">)</span>, </span>
<span>    count_marks_x <span class="op">=</span> <span class="fl">10</span>,</span>
<span>    count_marks_y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">20</span>, <span class="fl">30</span><span class="op">)</span>,</span>
<span>    highlight<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AAGTCCAGTACTATCGTACTA"</span>, <span class="st">"AAGTCCAGTACTGTAGCTACTA"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="UMI_VDJ_Barcode_files/figure-html/unnamed-chunk-12-1.png" width="480"></p>
</div>
</div>
<div class="section level2">
<h2 id="miscellaneous">Miscellaneous<a class="anchor" aria-label="anchor" href="#miscellaneous"></a>
</h2>
<p>We provides functions to transform the barcode information in
<code>BarcodeObj</code> to more general R data types.</p>
<div class="section level3">
<h3 id="sample-names">Sample names<a class="anchor" aria-label="anchor" href="#sample-names"></a>
</h3>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_names.html">bc_names</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "781_mixa" "781_mixb"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="output-to-data-frame">Output to data.frame<a class="anchor" aria-label="anchor" href="#output-to-data-frame"></a>
</h3>
<p><code>bc_2df</code> function uses the barcode and count info in the
<code>cleanBc</code> slot, outputs a data.frame contains: - barcode_seq:
barcode sequence - sample_name - count: reads or UMI count</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_2df.html">bc_2df</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt;   sample_name                      barcode_seq count</span></span>
<span><span class="co">#&gt; 1    781_mixa         AAGTCCAGTATCGTTACGCTACTA    53</span></span>
<span><span class="co">#&gt; 2    781_mixa           AAGTCCAGTACTGTAGCTACTA    50</span></span>
<span><span class="co">#&gt; 3    781_mixa            AAGTCCAGTACTATCGTACTA     6</span></span>
<span><span class="co">#&gt; 4    781_mixa              AAGTCCATCGTAGCTACTA    13</span></span>
<span><span class="co">#&gt; 5    781_mixa AAGTCCAGTTCTACTATCGTTACGAGCTACTA     4</span></span>
<span><span class="co">#&gt; 6    781_mixb           AAGTCCAGTACTGTAGCTACTA    37</span></span>
<span><span class="co">#&gt; 7    781_mixb         AAGTCCAGTATCGTTACGCTACTA    43</span></span>
<span><span class="co">#&gt; 8    781_mixb              AAGTCCATCGTAGCTACTA    20</span></span>
<span><span class="co">#&gt; 9    781_mixb AAGTCCAGTTCTACTATCGTTACGAGCTACTA     7</span></span></code></pre></div>
<p>Or if you prefer <code>data.table</code></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_2df.html">bc_2dt</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt;    sample_name                      barcode_seq count</span></span>
<span><span class="co">#&gt; 1:    781_mixa         AAGTCCAGTATCGTTACGCTACTA    53</span></span>
<span><span class="co">#&gt; 2:    781_mixa           AAGTCCAGTACTGTAGCTACTA    50</span></span>
<span><span class="co">#&gt; 3:    781_mixa            AAGTCCAGTACTATCGTACTA     6</span></span>
<span><span class="co">#&gt; 4:    781_mixa              AAGTCCATCGTAGCTACTA    13</span></span>
<span><span class="co">#&gt; 5:    781_mixa AAGTCCAGTTCTACTATCGTTACGAGCTACTA     4</span></span>
<span><span class="co">#&gt; 6:    781_mixb           AAGTCCAGTACTGTAGCTACTA    37</span></span>
<span><span class="co">#&gt; 7:    781_mixb         AAGTCCAGTATCGTTACGCTACTA    43</span></span>
<span><span class="co">#&gt; 8:    781_mixb              AAGTCCATCGTAGCTACTA    20</span></span>
<span><span class="co">#&gt; 9:    781_mixb AAGTCCAGTTCTACTATCGTTACGAGCTACTA     7</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="output-to-matrix">Output to matrix<a class="anchor" aria-label="anchor" href="#output-to-matrix"></a>
</h3>
<p><code>bc_2matrix</code> uses barcode and count information in
<code>cleanBc</code> slot to create reads count or UMI count matrix,
with barcodes in rows and samples in columns.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_2df.html">bc_2matrix</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                  X781_mixa X781_mixb</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTATCGTACTA                    6         0</span></span>
<span><span class="co">#&gt; AAGTCCAGTACTGTAGCTACTA                  50        37</span></span>
<span><span class="co">#&gt; AAGTCCAGTATCGTTACGCTACTA                53        43</span></span>
<span><span class="co">#&gt; AAGTCCAGTTCTACTATCGTTACGAGCTACTA         4         7</span></span>
<span><span class="co">#&gt; AAGTCCATCGTAGCTACTA                     13        20</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="more">More<a class="anchor" aria-label="anchor" href="#more"></a>
</h3>
<p>You can use:</p>
<ul>
<li>
<code>+</code>: to combine two BarcodeObj objects.</li>
<li>
<code>-</code>: to remove barcodes in a black list.</li>
<li>
<code>*</code>: only include barcodes in a white list.</li>
</ul>
<p>For examples:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">bc_obj</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Join two samples with different barcodes </span></span>
<span><span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test1"</span><span class="op">]</span> <span class="op">+</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AAAG"</span>, <span class="st">"test1"</span><span class="op">]</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 6 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AAAG", "test1"]  barcode_read_count.bc_obj["AAAG", "test1"]  depth_cutoff.bc_obj["AAAG", "test1"]</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 1 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 6 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 1 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 2 barcodes</span></span>
<span></span>
<span><span class="co"># Join two samples with shared barcodes</span></span>
<span><span class="va">bc_obj_join</span> <span class="op">&lt;-</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test1"</span><span class="op">]</span> <span class="op">+</span> <span class="va">bc_obj</span><span class="op">[</span><span class="st">"AGAG"</span>, <span class="st">"test1"</span><span class="op">]</span></span>
<span><span class="co">#&gt; Warning in merge.data.frame(metadata_x, metadata_y, by = 0, all = TRUE, :</span></span>
<span><span class="co">#&gt; column names 'raw_read_count.bc_obj["AGAG", "test1"]',</span></span>
<span><span class="co">#&gt; 'barcode_read_count.bc_obj["AGAG", "test1"]', 'depth_cutoff.bc_obj["AGAG",</span></span>
<span><span class="co">#&gt; "test1"]' are duplicated in the result</span></span>
<span><span class="va">bc_obj_join</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 6 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 1 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 3 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 1 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 1 barcodes</span></span>
<span></span>
<span><span class="co"># In this case, the shared barcodes are not merged.</span></span>
<span><span class="co"># Applying bc_cure_depth() to merge them.</span></span>
<span><span class="fu"><a href="../reference/bc_cure_depth.html">bc_cure_depth</a></span><span class="op">(</span><span class="va">bc_obj_join</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 7 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count.bc_obj["AGAG", "test1"]  barcode_read_count.bc_obj["AGAG", "test1"]  depth_cutoff.bc_obj["AGAG", "test1"]  raw_read_count.bc_obj["AGAG", "test1"].1  barcode_read_count.bc_obj["AGAG", "test1"].1  depth_cutoff.bc_obj["AGAG", "test1"].1  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 1 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 3 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 1 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 1 barcodes</span></span>
<span></span>
<span><span class="co"># Remove barcodes</span></span>
<span><span class="va">bc_obj</span> <span class="op">-</span> <span class="st">"AAAG"</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 3 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 7 Tags</span></span>
<span><span class="co">#&gt;     In sample $test2 there are: 7 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 3 barcodes</span></span>
<span><span class="co">#&gt;     In sample $test2 there are: 4 barcodes</span></span>
<span></span>
<span><span class="co"># Select barcodes in white list</span></span>
<span><span class="va">bc_obj</span> <span class="op">*</span> <span class="st">"AAAG"</span></span>
<span><span class="co">#&gt; Bonjour le monde, This is a BarcodeObj.</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; It contains: </span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @metadata: 3 field(s) available:</span></span>
<span><span class="co">#&gt; raw_read_count  barcode_read_count  depth_cutoff</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @messyBc: 2 sample(s) for raw barcodes:</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 3 Tags</span></span>
<span><span class="co">#&gt;     In sample $test2 there are: 2 Tags</span></span>
<span><span class="co">#&gt; ----------</span></span>
<span><span class="co">#&gt; @cleanBc: 2 samples for cleaned barcodes</span></span>
<span><span class="co">#&gt;     In sample $test1 there are: 1 barcodes</span></span>
<span><span class="co">#&gt;     In sample $test2 there are: 1 barcodes</span></span></code></pre></div>
<p>What’s more, by combining several functions, it is possible to
accomplish more complex task. In the following example, a barcode from
sample “781_mixa” is selected , then output the result in
<code>data.frame</code> format.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/bc_2df.html">bc_2df</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">[</span><span class="fu"><a href="../reference/bc_barcodes.html">bc_barcodes</a></span><span class="op">(</span><span class="va">bc_obj_umi_sub</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="st">"781_mixa"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;   sample_name              barcode_seq count</span></span>
<span><span class="co">#&gt; 1    781_mixa AAGTCCAGTATCGTTACGCTACTA    53</span></span>
<span>                  <span class="co">## 1. Use `bc_barcodes` to pull out all the barcodes in two</span></span>
<span>                  <span class="co">##    samples, and choose the fist barcode.</span></span>
<span>       <span class="co">## 2. Select the barcode got in step 1, and the sample named "781_mixa".</span></span>
<span><span class="co">## 3. Convert the BarcodeObj object to a data.frame. </span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: macOS 14.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Paris</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] magrittr_2.0.3    CellBarcode_1.7.0 ggplot2_3.4.2     data.table_1.14.8</span></span>
<span><span class="co">#&gt; [5] BiocStyle_2.28.0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0            farver_2.1.1               </span></span>
<span><span class="co">#&gt;  [3] dplyr_1.1.2                 Biostrings_2.68.1          </span></span>
<span><span class="co">#&gt;  [5] bitops_1.0-7                fastmap_1.1.1              </span></span>
<span><span class="co">#&gt;  [7] RCurl_1.98-1.12             GenomicAlignments_1.36.0   </span></span>
<span><span class="co">#&gt;  [9] egg_0.4.5                   digest_0.6.31              </span></span>
<span><span class="co">#&gt; [11] lifecycle_1.0.3             compiler_4.3.0             </span></span>
<span><span class="co">#&gt; [13] rlang_1.1.1                 sass_0.4.6                 </span></span>
<span><span class="co">#&gt; [15] tools_4.3.0                 utf8_1.2.3                 </span></span>
<span><span class="co">#&gt; [17] yaml_2.3.7                  knitr_1.43                 </span></span>
<span><span class="co">#&gt; [19] labeling_0.4.2              S4Arrays_1.0.4             </span></span>
<span><span class="co">#&gt; [21] interp_1.1-4                DelayedArray_0.26.3        </span></span>
<span><span class="co">#&gt; [23] plyr_1.8.8                  RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt; [25] ShortRead_1.58.0            BiocParallel_1.34.2        </span></span>
<span><span class="co">#&gt; [27] withr_2.5.0                 purrr_1.0.1                </span></span>
<span><span class="co">#&gt; [29] hwriter_1.3.2.1             BiocGenerics_0.46.0        </span></span>
<span><span class="co">#&gt; [31] desc_1.4.2                  grid_4.3.0                 </span></span>
<span><span class="co">#&gt; [33] stats4_4.3.0                fansi_1.0.4                </span></span>
<span><span class="co">#&gt; [35] latticeExtra_0.6-30         Ckmeans.1d.dp_4.3.4        </span></span>
<span><span class="co">#&gt; [37] colorspace_2.1-0            scales_1.2.1               </span></span>
<span><span class="co">#&gt; [39] SummarizedExperiment_1.30.2 cli_3.6.1                  </span></span>
<span><span class="co">#&gt; [41] rmarkdown_2.22              crayon_1.5.2               </span></span>
<span><span class="co">#&gt; [43] ragg_1.2.5                  generics_0.1.3             </span></span>
<span><span class="co">#&gt; [45] cachem_1.0.8                stringr_1.5.0              </span></span>
<span><span class="co">#&gt; [47] zlibbioc_1.46.0             parallel_4.3.0             </span></span>
<span><span class="co">#&gt; [49] BiocManager_1.30.21         XVector_0.40.0             </span></span>
<span><span class="co">#&gt; [51] matrixStats_1.0.0           vctrs_0.6.3                </span></span>
<span><span class="co">#&gt; [53] Matrix_1.6-0                jsonlite_1.8.5             </span></span>
<span><span class="co">#&gt; [55] bookdown_0.34               IRanges_2.34.0             </span></span>
<span><span class="co">#&gt; [57] S4Vectors_0.38.1            systemfonts_1.0.4          </span></span>
<span><span class="co">#&gt; [59] jpeg_0.1-10                 jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [61] glue_1.6.2                  pkgdown_2.0.7              </span></span>
<span><span class="co">#&gt; [63] codetools_0.2-19            stringi_1.7.12             </span></span>
<span><span class="co">#&gt; [65] gtable_0.3.3                GenomeInfoDb_1.36.0        </span></span>
<span><span class="co">#&gt; [67] deldir_1.0-9                GenomicRanges_1.52.0       </span></span>
<span><span class="co">#&gt; [69] munsell_0.5.0               tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [71] pillar_1.9.0                htmltools_0.5.5            </span></span>
<span><span class="co">#&gt; [73] GenomeInfoDbData_1.2.10     R6_2.5.1                   </span></span>
<span><span class="co">#&gt; [75] textshaping_0.3.6           Rdpack_2.4                 </span></span>
<span><span class="co">#&gt; [77] rprojroot_2.0.3             evaluate_0.21              </span></span>
<span><span class="co">#&gt; [79] Biobase_2.60.0              lattice_0.21-8             </span></span>
<span><span class="co">#&gt; [81] highr_0.10                  rbibutils_2.2.13           </span></span>
<span><span class="co">#&gt; [83] png_0.1-8                   Rsamtools_2.16.0           </span></span>
<span><span class="co">#&gt; [85] memoise_2.0.1               bslib_0.5.0                </span></span>
<span><span class="co">#&gt; [87] Rcpp_1.0.10                 gridExtra_2.3              </span></span>
<span><span class="co">#&gt; [89] xfun_0.39                   fs_1.6.2                   </span></span>
<span><span class="co">#&gt; [91] MatrixGenerics_1.12.2       pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Wenjie Sun, Anne-Marie Lyne, Leila Perie.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
